components/use-click-away/__tests__/click-away.test.tsx

```diff
@@ -34,4 +34,23 @@ describe('UseClickAway', () => {
 
     expect(errorSpy).not.toHaveBeenCalled()
   })
+
+  it('should update handler reference', async () => {
+    const ref = React.createRef() as React.MutableRefObject<HTMLDivElement>
+    ref.current = document.createElement('div')
+    const handler1 = jest.fn()
+    const handler2 = jest.fn()
+    let isFirstRender = true
+    const { rerender } = renderHook(() => {
+      const handler = isFirstRender ? handler1 : handler2
+      useClickAway(ref, handler)
+      isFirstRender = false
+    })
+    simulateNativeClick(document.body)
+    expect(handler1).toHaveBeenCalled()
+
+    rerender()
+    simulateNativeClick(document.body)
+    expect(handler2).toHaveBeenCalled()
```

Comment:
`handler1` should not be called?

Reply:
`handler1` and `handler2` should be called once each in this test.

Reply:
This assertion is mainly ensure `handler2` has been called.